docs/spec/sources.md

```diff
@@ -362,5 +376,39 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+
+Examples:
+
+- `hello.pingsources.sources.knative.dev`
+- `myqueue.rabbitmqsources.sources.knative.dev`
+
+It is allowed to specify an alternative value in
+[`spec.ceOverrides.extensions`](#duckspec), in which case the `knsource` value
+can be of any shape.
+
+When the `knsource` CloudEvent extension already exists in the CloudEvent being
```

Comment:
This is very confusing.

Reply:
What do you mean? Not sure how trace headers can help.

Reply:
it seems like you want to filter based on the path the event took to the broker and if we could record that data and link the trace span with a source instance, it might get what you want.

Reply:
I think one of the use-cases is the original [faas scenario](https://docs.google.com/document/d/1DpiSL2dUcYS2n7yXOIG5LJwyIC1lY9q_W8-56U1SvKM/edit?hl=en) where a trigger and the source are instantiated and chained together in one go.